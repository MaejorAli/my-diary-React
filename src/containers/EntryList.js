import React from 'react';

const dateType = (date) => {
  const parsedDate = new Date(date);
  const months = ['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
  const entryDate = `${months[parsedDate.getMonth()]} ${parsedDate.getDate()}, ${parsedDate.getFullYear()}`;
  return entryDate;
};

const EntryList = (props) => (
      <>
        <div className="entry-description" id="entries">
          <ul>
            <li>
              <h2>{props.title}</h2>
            </li>

            <li>
              <p>{dateType(props.createdat)}</p>
            </li>
            <li>
              <button type="button" id="entries" onClick={() => props.history.push(`/entries/${props.id}`)}>Read More</button>
            </li>
          </ul>
        </div>
      </>
);

export default EntryList;
