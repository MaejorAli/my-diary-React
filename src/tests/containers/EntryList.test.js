import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import EntryList from '../../containers/EntryList';


const setup = (overrideProps) => {
  const props = {
    signupUser: jest.fn(),
    signin: jest.fn(),
    PublishEntry: jest.fn(),
    getAllEntries: jest.fn(),
    GetEntries: jest.fn(),
    entries: [],
    entry: {},
    match: {
      params: { id: 1 },
    },
    deleteEntry: jest.fn(),
    deleteEntryReducer: jest.fn(),
    history: { push: jest.fn() },
  };
  const wrapped = shallow(<EntryList {...props} {...overrideProps} />);
  return { props, wrapped };
};

describe('Get An Entry container', () => {
  describe('render features', () => {
    test('view should render as expected', () => {
      const { wrapped } = setup();
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
