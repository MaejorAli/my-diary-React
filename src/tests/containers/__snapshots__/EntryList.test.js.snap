// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get An Entry container render features view should render as expected 1`] = `
<Fragment>
  <div
    className="entry-description"
    id="entries"
  >
    <ul>
      <li>
        <h2 />
      </li>
      <li>
        <p>
          undefined NaN, NaN
        </p>
      </li>
      <li>
        <button
          id="entries"
          onClick={[Function]}
          type="button"
        >
          Read More
        </button>
      </li>
    </ul>
  </div>
</Fragment>
`;
